main.rb

```diff
@@ -0,0 +1,118 @@
+# rubocop:disable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity, Metrics/ModuleLength
+module Enumerable
+  def my_each
+    i = 0
+    while i < to_a.length
+      yield(to_a[i])
+      i += 1
+    end
+  end
```

Comment:
- [ ] returns Enumerator if no block is given

- [ ] when self is an array or a range it returns the array or range itself after calling the given block once for each element in self. You can solve this by returning self at the end.